src/peer_manager.rs

```diff
@@ -1237,44 +1246,49 @@ impl PeerManager {
             .collect()
     }
 
-    /// Returns the `PeerId`s and names of all the peers held in the RT, and whether they are
-    /// connected via a tunnel or not.
-    pub fn get_routing_peer_details(&self) -> Vec<(PeerId, XorName, bool)> {
-        self.routing_table
-            .iter()
-            .filter_map(|name| -> Option<(PeerId, XorName, bool)> {
-                let peer = match self.peer_map.get_by_name(name) {
-                    Some(peer) => peer,
-                    None => {
-                        error!("{:?} Have {} in RT, but have no entry in peer_map for it.",
+    /// Returns all syncing peer's `PeerId`s, names and whether connected via a tunnel or not;
+    /// together with all out-of-sync peer's names and peer_ids if have.
+    pub fn get_routing_peer_details(&self) -> PeerDetails {
```

Comment:
Let's split this up into two methods, one returning the known peers and one returning the unknown routing table names. Let's try to avoid making this epic inline closure even more complex. 😬